htdocs/core/actions_massactions.inc.php

```diff
@@ -682,17 +686,17 @@
 	if (! $error && $validate_invoices)
 	{
 		$massaction = $action = 'builddoc';
-		foreach($TAllFact as &$objecttmp)
+		foreach($TAllFact as &$object)
```

Comment:
Do not use $object as a temporary object into a loop because this erase the $object that is used in a global context page that call the inc.

Reply:
When mass generating invoices, in [actions_massactions.inc.php](https://github.com/Dolibarr/dolibarr/blob/8.0/htdocs/core/actions_massactions.inc.php), $object is an instance of Commande. But, in [actions_builddoc.inc.php](https://github.com/Dolibarr/dolibarr/blob/8.0/htdocs/core/actions_builddoc.inc.php), $object must be an instance of Facture.

How to solve it ?

Reply:
I don't understand. We are in a loop
foreach($TAllFact as &$objecttmp)

so $objecttmp here is an instance of what we have into the array $TAllFact

Reply:
Ok. I changed my PR.